From 6f819fda93986f7a07c646bcad3587411ac2b00e Mon Sep 17 00:00:00 2001
From: Peter Polonec <polonec.devel@pm.me>
Date: Wed, 26 Aug 2026 14:30:54 +0200
Subject: [PATCH 128/129] openssl: update to 3.5.8

Update OpenSSL to 3.5.8, a security patch release.

This fixes:
- CVE-2026-18798
- CVE-2026-63072
- CVE-2026-63076
- CVE-2026-14456
- CVE-2026-14457
- CVE-2026-54874
- CVE-2026-63073
- CVE-2026-63074
- CVE-2026-63075
- CVE-2026-75803

Link: https://openssl-library.org/news/secadv/20260825.txt
Signed-off-by: Peter Polonec <polonec.devel@pm.me>
Link: https://github.com/openwrt/openwrt/pull/24915
(cherry picked from commit 78f136f03d144f747c03578a61d42925ca9eee2c)
Link: https://github.com/openwrt/openwrt/pull/25326
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit 7ac5eaf362f7d42fd68537173cc1f102786876a7)
---
 package/libs/openssl/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 76a04b4b50..b4675a25a2 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openssl
-PKG_VERSION:=3.5.7
+PKG_VERSION:=3.5.8
 PKG_RELEASE:=1
 PKG_BUILD_FLAGS:=no-mips16 gc-sections no-lto
 
@@ -21,7 +21,7 @@ PKG_SOURCE_URL:= \
 	https://www.openssl.org/source/old/$(PKG_BASE)/ \
 	https://github.com/openssl/openssl/releases/download/$(PKG_NAME)-$(PKG_VERSION)/
 
-PKG_HASH:=a8c0d28a529ca480f9f36cf5792e2cd21984552a3c8e4aa11a24aa31aeac98e8
+PKG_HASH:=a8f84a39918ec6415ce765d9b429d313ba97b8143169c172e734b9514464f5b2
 
 PKG_LICENSE:=Apache-2.0
 PKG_LICENSE_FILES:=LICENSE.txt
-- 
2.34.1

